### PR TITLE
Add support for type and user_id queries on get_subscriptions

### PR DESCRIPTION
**Pull request by PredaaA**
This PR adds support for `type` and `user_id` queries to EventSub get_subscriptions.
https://dev.twitch.tv/docs/api/reference/#get-eventsub-subscriptions

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
    - [x] I have updated the changelog with a quick recap of my changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by chillymosh** [comment]
Just to note, the filters are mutually exclusive so if more than one are sent it will probably fail with a 400 error.

The typings should probably be `Optional` since they are set to None and are also not required.

I'll let @IAmTomahawkx confirm this as he worked on Eventsub.

File: twitchio/ext/eventsub/server.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ async def delete_all_active_subscriptions(self):
         for subscription_id in active_subscriptions:
             await self.delete_subscription(subscription_id)
 
-    async def get_subscriptions(self, status: str = None):
+    async def get_subscriptions(self, status: str = None, sub_type: str = None, user_id: str = None):
```
Comment (IAmTomahawkx) [inline twitchio/ext/eventsub/server.py:55]: both of these are Optional, user_id should be Optional[int] at a user facing level to remain consistent with the rest of the lib. status should technically also be Optional[str] to actually pass type checking.

**Review by IAmTomahawkx — CHANGES_REQUESTED** [review]
You'll need to ensure that only one of these is passed as chilly mentioned, probably best to do it at the http level. Other than that just some typing nitpicks (a lot of the lib is typed like this, unfortunately how I did it is actually wrong)

File: twitchio/ext/eventsub/server.py (hunk 1 above)
Comment (IAmTomahawkx) [reply twitchio/ext/eventsub/server.py:55]: don't forget to cast back to `str` on the user_id in the underlying method

**Comment by PredaaA** [comment]
Just pushed requested changes.

**Review by IAmTomahawkx — APPROVED** [review]
LGTM

**Comment by IAmTomahawkx** [comment]
@PredaaA can you fix the merge conflicts so i can merge this? thanks!

**Comment by PredaaA** [comment]
Yup, done!

**Pull request merged**